docs/proposals/security-context-constraints.md

```diff
@@ -0,0 +1,292 @@
+type SecurityContextConstraints struct {
+  // Type distinguishes the type of SecurityContextConstraints object.  Acceptable values are
+  // INFRA, USER, DEFAULT_USER, SERVICE_ACCOUNT, DEFAULT_SERVICE_ACCOUNT
+  Type string
+  // AllowPrivilegedContainer determines if a container can request to be run as privileged.
+  AllowPrivilegedContainer bool
+  // AllowedCapabilities is a list of capabilities that can be requested to add to the container.
+  AllowedCapabilities []CapabilityType
+  // AllowHostDirVolumePlugin determines if the policy allow containers to use the HostDir volume plugin
+  AllowHostDirVolumePlugin bool
+  // The users who have permissions to use this security context constraints
+  AllowedUsers []string
+  // The service accounts that have permission to use this security context constraints
+  AllowedServiceAccounts []NamespacedName
+}
+
+type SecurityContextConstraintsAllocator interface {
+  // Create a SecurityContext based on the given constraints
+  CreateSecurityConstraints(constraints *api.SecurityContextConstraints) api.SecurityContext
+  // Ensure a container's SecurityContext is in compliance with the given constraints
+  ValidateSecurityConstraints(container *api.Container, constraints *api.SecurityContextConstraints) fielderrors.ValidationErrorList
+}
+
+// implements the SecurityContextConstraintsAllocator interface
+type SimpleSecurityContextConstraintsAllocator struct {
+    // SELinuxContext is the strategy that will dictate what labels will be set in the SecurityContext.
+    SELinuxContext SELinuxContextStrategy
+    // RunAsUser is the strategy that will dictate what RunAsUser is used in the SecurityContext.
+    RunAsUser RunAsUserStrategy
+}
+
+const (
+  // container must have SELinux labels of X applied.
+  SELinuxStrategyMustRunAs SELinuxContextStrategyType = "MustRunAs"
+  // container may make requests for any SELinux context labels.
+  SELinuxStrategyRunAsAny SELinuxContextStrategyType = "RunAsAny"
+  // containers must run with the default settings, their requests are ignored
+  SELinuxStrategyRunAsDefault SELinuxContextStrategyType = "RunAsDefault"
+
+  // container must run as a particular uid.
+  RunAsUserStrategyMustRunAs RunAsUserStrategyType = "MustRunAs"
+  // container must run as a non-root uid
+  RunAsUserStrategyMustRunAsNonRoot RunAsUserStrategyType = "MustRunAsNonRoot"
+  // container may make requests for any uid.
+  RunAsUserStrategyRunAsAny RunAsUserStrategyType = "RunAsAny"
+  // containers must run with the default settings, their requests are ignored
+  RunAsUserStrategyRunAsDefault RunAsUserStrategyType = "RunAsDefault"
+)
+```
+
+#### Security Context Constraints Lifecycle
+
+As reusable objects in the root scope, security context constraints follow the lifecycle of the
+cluster itself.  Maintenance of constraints such as adding, assigning, or changing them is the
+responsibility of the cluster administrator but is assisted by automation within the cluster.  For
```

Comment:
"but is assisted by automation within the cluster" isn't well-defined... drop it?